src/transactions/transaction_builder/transaction_builder.ts

```diff
@@ -126,6 +131,111 @@ export function generateTransactionPayload(args: GenerateTransactionPayloadData)
+  aptosConfig: AptosConfig,
+): Promise<TransactionPayload> {
+  const funcNameParts = args.function.split("::");
+  const moduleAddress = funcNameParts[0];
+  const moduleName = funcNameParts[1];
+  const functionName = funcNameParts[2];
+
+  // We fetch the entry function ABI, and then pretend that we already had the ABI
+  const functionAbi = await fetchEntryFunctionAbi(moduleAddress, moduleName, functionName, aptosConfig);
+  return generateTransactionPayloadWithABI(args, functionAbi);
+}
+
+export function generateTransactionPayloadWithABI(
+  args: EntryFunctionDataWithABI,
+  functionAbi: MoveFunction,
+): TransactionPayloadEntryFunction;
+export function generateTransactionPayloadWithABI(
+  args: MultiSigDataWithABI,
+  functionAbi: MoveFunction,
+): TransactionPayloadMultisig;
+export function generateTransactionPayloadWithABI(
+  args: GenerateTransactionPayloadDataWithABI,
+  functionAbi: MoveFunction,
+): TransactionPayload;
+export function generateTransactionPayloadWithABI(
+  args: GenerateTransactionPayloadDataWithABI,
+  functionAbi: MoveFunction,
+): TransactionPayload {
+  const funcNameParts = args.function.split("::");
+  const moduleAddress = funcNameParts[0];
+  const moduleName = funcNameParts[1];
+  const functionName = funcNameParts[2];
+
+  // We have to find the first non-signer arg for arguments
+  const startAbiIndex = findFirstNonSignerArg(functionAbi);
+  let abiIndex = startAbiIndex;
+
+  // Ensure that all type arguments are typed properly
+  const typeArguments = typeTagConversion(args.typeArguments);
+
+  // Check the type argument count against the ABI
+  if (typeArguments.length !== functionAbi?.generic_type_params.length) {
+    throw new Error(
+      `Type argument count mismatch, expected ${functionAbi.generic_type_params.length}, received ${typeArguments.length}`,
+    );
+  }
+
+  // Check all BCS types, and convert any non-BCS types
+  const functionArguments: Array<EntryFunctionArgumentTypes> = [];
+  for (let i = 0; i < args.functionArguments.length; i += 1) {
+    const arg = args.functionArguments[i];
+    convertArgument(args.function, functionAbi, arg, abiIndex, i, startAbiIndex);
+    abiIndex += 1;
+  }
+
+  // Check that all arguments are accounted for
+  if (abiIndex !== functionAbi.params.length) {
+    throw new Error(
+      // eslint-disable-next-line max-len
+      `Too few arguments for '${moduleAddress}::${moduleName}::${functionName}', expected ${
+        functionAbi.params.length - startAbiIndex
+      } but got ${abiIndex - startAbiIndex}`,
+    );
+  }
+
+  // Generate entry function payload
+  const entryFunctionPayload = EntryFunction.build(
+    `${moduleAddress}::${moduleName}`,
+    functionName,
+    typeArguments,
+    functionArguments,
+  );
+
+  // Send it as multi sig if it's a multisig payload
+  if ("multisigAddress" in args) {
+    let multisigAddress: AccountAddress;
+    if (typeof args.multisigAddress === "string") {
```

Comment:
I wonder if it's cleaner to also offer this as a library function - toAddress(String | AccountAddress)

Reply:
I will probably just adjust the AccountAddress later to do that.

There's a bunch of little things like this that eat away at type conversion.

There are other pieces of the code that use HexInput for AccountAddress, and tbh no one is going to be using a byte array for AccountAddress.